src/wallet/scriptpubkeyman.cpp

```diff
@@ -1501,7 +1501,48 @@ void LegacyScriptPubKeyMan::SetType(OutputType type, bool internal) {}
 
 bool DescriptorScriptPubKeyMan::GetNewDestination(const OutputType type, CTxDestination& dest, std::string& error)
 {
-    return false;
+    // Returns true if we have private keys (encrypted or not) or we have cached keys. But we may still be unable to get addresses, these conditions are caught later.
+    if (!CanGetAddresses(internal)) {
+        error = "No addresses available";
+        return false;
+    }
+    {
+        LOCK(cs_desc_man);
+        if (m_wallet_descriptor.descriptor->IsSingleType() && type != address_type) {
+            throw std::runtime_error(std::string(__func__) + ": Types are inconsistent");
+        }
+
+        TopUp();
+
+        // Get the scriptPubKey from the descriptor
+        FlatSigningProvider out_keys;
+        std::vector<CScript> scripts_temp;
+        if (m_wallet_descriptor.range_end <= m_max_cached_index && !TopUp(1)) {
```

Comment:
`TopUp(1) generates the next cache item if we have run out of cached things. `ExpandFromCache` can then use that cached thing. Otherwise `TopUp(1)` is a no-op.